classes/class04/homework/routes/base.js

```diff
@@ -0,0 +1,99 @@
+var schema = require('../models/schema');
+
+exports.home = function (req, res) {
+    res.render('index');
+};
+
+exports.ingredients = function (req, res) {
+    schema.Ingredient.find({}, function (err, data) {
+        res.render('ingredients', {ingredients: data});
+    })
+}
+
+exports.order = function (req, res) {
+    schema.Ingredient.find({}, function (err, data) {
+        if (err) {
+            console.log("Failed to get list of ingredients");
+            res.status(500, {error: "Failed to get list of ingredients " + err});
```

Comment:
res.status() does not end the response (this would cause the server to never respond to the client).
Also, res.status() only takes a status number, not a data object.
http://expressjs.com/api.html#res.status